src/Planners/src/TimeVaryingDCMPlanner.cpp

```diff
@@ -392,10 +377,50 @@ struct TimeVaryingDCMPlanner::Impl
                 feetAreInSamePlane = computeConstraintElementsECMP(*contactPhaseListIt,
                                                                    ecmpConstraintA,
                                                                    ecmpConstraintB);
+            }
+
+            if (!feetAreInSamePlane)
+            {
+                this->opti.subject_to(
+                    casadi::MX::mtimes(ecmpConstraintA,
+                                       casadi::MX::vertcat(this->optiFunctions.ecmp(
+                                           {omega(Sl(), k), vrp(Sl(), k), omegaDot(Sl(), k)})))
+                    <= ecmpConstraintB);
+            } else
+            {
+                this->opti.subject_to(casadi::MX::mtimes(ecmpConstraintA, vrp(Sl(0, 2), k))
+                                      <= ecmpConstraintB);
+
+                this->opti.subject_to(
+                    vrp(2, k) - 9.81 / (casadi::MX::pow(omega(Sl(), k), 2) - omegaDot(Sl(), k))
+                    == contactPhaseListIt->activeContacts.begin()->second->pose.translation()[2]);
```

Comment:
Can we define a placeholder for -9.81?

Reply:
If we define a constant somewhere, let's use the internationally agreed standard CODATA2018  of 9.80665 https://physics.nist.gov/cgi-bin/cuu/Value?gn

Reply:
Mmmh ok where?

Reply:
I would expose this value in some way. Often when debugging I personally use and simulate with `-10.0` and it would be great at this point if this parameter can be exposed to the user. Note that the initial omega included in the initial state and passed to `setInitialState` is typically computed by the users using their value of `g`. If, then, internally another value is hardcoded, it could get funky results.

As a matter of facts, I remember in the past changing the value to something different than 9.81 and the optimization was failing. I didn't delve into further details at that time. I'm not saying it was related to it, but it rang a bell in my memory.

Reply:
Yes, those are two different aspects: 
* It would be great to have `9.80665` define somewhere like `include/BipedalLocomotion/Math/Constants.h` or something similar.
* `g` in this case is a parameter of the `TimeVaryingDCMPlanner` , and if not specified (if this is supported, not sure) it should be fallback to the previous define constant.

Reply:
We can add it as an optional parameter. What do you think? In case the parameter is not passed we can use the standard suggested by @traversaro.

Now we should define where to put the constants. We can create another component.

Reply:
I think we hit "send" together. :D

Reply:
This would be ideal!

Reply:
> As a matter of facts, I remember in the past changing the value to something different than 9.81 and the optimization was failing. I didn't delve into further details at that time. I'm not saying it was related to it, but it rang a bell in my memory.

In my experience, I have faced such issues as well. While integrating system with 9.80665 in one place and using 9.81 in another place, by choosing a big value for sampling time introduces errors easily into the estimator/filter.

Reply:
Ops I've just noticed https://github.com/dic-iit/bipedal-locomotion-framework/pull/208#discussion_r582664434  I created a new component. Now I ever the changes to follow what @traversaro suggested in https://github.com/dic-iit/bipedal-locomotion-framework/pull/208#discussion_r582664434

Reply:
I think also what you did is perfect, no need to change to do exactly what I said.

Reply:
I opened an issue https://github.com/dic-iit/bipedal-locomotion-framework/issues/210 to propagate the use of gravity constant in the rest of the codebase. No need to do it in this PR tough.

Reply:
Here I am! I introduced `BipedalLocomotion::Math::StandardAccelerationOfGravitation` https://github.com/dic-iit/bipedal-locomotion-framework/pull/208/commits/a4af4be82df8e6ee72c958c4ad3e32d71c513b41 I exposed it in python https://github.com/dic-iit/bipedal-locomotion-framework/pull/208/commits/2e625fd0971149f1ab3c6aba7144d02f12cbe9f9

I finally introduced the `gravity` parameter in `TimeVaryingDCMPlanner` https://github.com/dic-iit/bipedal-locomotion-framework/pull/208/commits/a24da91c8b4af089deb2366824cf4c8485422756. It's optional and if the user do not set it `BipedalLocomotion::Math::StandardAccelerationOfGravitation` is used